File: 0002-add-two-numbers/0002-add-two-numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    dummy := &ListNode{Val: 0}
    current := dummy
    curry := 0
    firstList := l1
    secondList := l2
    
    for firstList != nil || secondList != nil || curry != 0 {
        var value1 , value2 int
        if firstList != nil {
            value1 = firstList.Val
        } else {
            value1 = 0
        }
        
         if secondList != nil {
            value2 = secondList.Val
         } else {
            value2 = 0 
         }
        
        sum := value1 + value2 + curry
        
        newValue := sum % 10 
        
        newNode := &ListNode{Val: newValue}
        
        current.Next = newNode
        current = newNode
        
        curry = sum / 10 
        if firstList != nil {
                   firstList = firstList.Next  
        } else {
                    firstList = nil
        }
        
        if secondList != nil {
                   secondList = secondList.Next  
        } else {
                    secondList = nil
        }

        
    }
    return dummy.Next
    
}

File: 0002-add-two-numbers/0002-add-two-numbers.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} l1
 * @param {ListNode} l2
 * @return {ListNode}
 */
var addTwoNumbers = function(l1, l2) {
    let dummy = new ListNode(0)
    let pointer = dummy   
    let curry = 0

    let firstList  = l1
     let secondList = l2

    while (firstList !== null || secondList !== null || curry !== 0) {

        valueOne = firstList !== null ?  firstList.val : 0
        valueTwo = secondList !== null ? secondList.val : 0

        let sum = valueOne + valueTwo + curry

        let newValue = sum % 10
        let newNode = new ListNode(newValue)
        pointer.next = newNode
        pointer = newNode
        
        curry = Math.floor(sum / 10)

       
        firstList = firstList !== null ? firstList.next : null
       
        secondList = secondList !== null ? secondList.next : null
        
    } 
    return dummy.next
};